src/statistics.rs

```diff
@@ -0,0 +1,771 @@
+//! Delta Chat has an advanced option
+//! "Send statistics to the developers of Delta Chat".
+//! If this is enabled, a JSON file with some anonymous statistics
+//! will be sent to a bot once a week.
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use anyhow::{Context as _, Result, ensure};
+use deltachat_derive::FromSql;
+use pgp::types::PublicKeyTrait;
+use serde::Serialize;
+
+use crate::chat::{self, ChatId, ChatVisibility, MuteDuration, ProtectionStatus};
+use crate::config::Config;
+use crate::constants::Chattype;
+use crate::contact::{Contact, ContactId, Origin, import_vcard, mark_contact_id_as_verified};
+use crate::context::{Context, get_version_str};
+use crate::key::load_self_public_keyring;
+use crate::log::LogExt;
+use crate::message::{Message, Viewtype};
+use crate::securejoin::QrInvite;
+use crate::tools::{create_id, time};
+
+pub(crate) const STATISTICS_BOT_EMAIL: &str = "self_reporting@testrun.org";
+const STATISTICS_BOT_VCARD: &str = include_str!("../assets/statistics-bot.vcf");
+const SENDING_INTERVAL_SECONDS: i64 = 3600 * 24 * 7; // 1 week
+
+#[derive(Serialize)]
+struct Statistics {
```

Comment:
Maybe add some serial number here so that the bot can see if some statistics were lost?

Reply:
There are no more differential statistics anymore now